micro-rdk-nmea/src/parse_helpers/parsers.rs

```diff
@@ -0,0 +1,404 @@
+use bitvec::prelude::*;
+use std::{
+    marker::PhantomData,
+    rc::Rc,
+    sync::atomic::{AtomicUsize, Ordering},
+};
+
+use micro_rdk::common::sensor::GenericReadingsResult;
+
+use super::{
+    enums::NmeaEnumeratedField,
+    errors::{NmeaParseError, NumberFieldError},
+};
+
+/// Cursor over a byte slice of data. The cursor can be moved by either consuming
+/// child-instances of `DataRead` or calling the `advance` function
+pub struct DataCursor<'a> {
+    data: &'a [u8],
+    bit_position: Rc<AtomicUsize>,
```

Comment:
I don't understand why atomics are making an appearance.

Reply:
I had a reason, but I have since forgotten it. I'll just remove it, it's not like the rest of the code is thread-safe anyway

Reply:
Ok I've remembered again. Rust will not let me mutate the content of an `Rc` in the drop for `DataRead` because it doesn't consider it safe to mutate it when the reference count isn't 0. I have to get around this with the interior mutability of `AtomicUsize`